src/ui/public/courier/data_source/_normalize_sort_request.js

```diff
@@ -56,3 +56,19 @@ export default function normalizeSortRequest(config) {
     return normalized;
   }
 };
+
+function castSortType(type) {
```

Comment:
I would just add a comment, something like what is in the issue, `The elasticsearch API only supports sort scripts of type number and string, so cast ....`.  Then maybe some day down the line if es API changes and does support sort of those types, we'll know it's safe to remove this extra code.